third_party/move/move-vm/types/src/loaded_data/runtime_types.rs

```diff
@@ -1133,6 +1156,36 @@ impl TypeBuilder {
                     ability: ability.clone(),
                 }
             },
+            Function {
+                args,
+                results,
+                abilities,
+            } => {
+                let subs_elem = |count: &mut u64,
+                                 ty: &TriompheArc<Type>|
+                 -> PartialVMResult<TriompheArc<Type>> {
+                    Ok(TriompheArc::new(Self::apply_subst(
```

Comment:
This seems very wasteful: we take a an arc, deep copy + substitute it and then create yet another arc for each argument and each return